Mathlib/MeasureTheory/Decomposition/Exhaustion.lean

```diff
@@ -0,0 +1,207 @@
+/-
+Copyright (c) 2024 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne
+-/
+import Mathlib.MeasureTheory.Measure.Typeclasses
+
+/-!
+# Method of exhaustion
+
+If `μ, ν` are two measures with `ν` finite, then there exists a set `s` such that
+`μ` is sigma-finite on `s`, and for all sets `t ⊆ sᶜ`, either `ν t = 0` or `μ t = ∞`.
```

Comment:
I would expect that this statement generalizes readily when nu is S-finite (because it is then equivalent to a finite measure). What about defining `sigmaFiniteSetWRT` without any assumptions on nu as: if there exists a measurable set s such that (mu.restrict s is sigma-finite and for all t in s^c, either nu t = 0 or mu t = \infty), then pick such a set s, else pick the empty set. Then prove that for finite measures, it does what it should do (following exactly your proof), and then deduce the same for S-finite nu?